sqlgen/db.go

```diff
@@ -232,6 +250,64 @@ func (db *DB) checkColumnValuesAgainstLimits(ctx context.Context, query SQLQuery
 	return nil
 }
 
+type ExplainResultRow struct {
+	Id           int64
+	SelectType   string `sql:"select_type"`
+	Table        string
+	TypeColumn   string  `sql:"type"`
+	PossibleKeys *string `sql:"possible_keys"`
+	Key          *string
+	KeyLen       *string `sql:"key_len"`
+	Ref          *string
+	Rows         int64
+	Extra        *string `sql:"Extra"`
+}
+
+func parseExplainResults(resultRows *sql.Rows) (results []ExplainResultRow, err error) {
+	for resultRows.Next() {
+		var explainType ExplainResultRow
+		err := resultRows.Scan(
+			&explainType.Id, &explainType.SelectType, &explainType.Table, &explainType.TypeColumn, &explainType.PossibleKeys, &explainType.Key,
+			&explainType.KeyLen, &explainType.Ref, &explainType.Rows, &explainType.Extra)
+		if err != nil {
+			return nil, oops.Wrapf(err, "Failed to parse explain results")
+		}
+		results = append(results, explainType)
+	}
+	return results, nil
+}
+
+func (db *DB) runExplainQuery(ctx context.Context, clause string, args []interface{}) error {
+	// We run an explain first and panic if there's no index
+	explained := "EXPLAIN " + clause
+	res, err := db.QueryExecer(ctx).QueryContext(ctx, explained, args...)
+	if err != nil {
+		return oops.Wrapf(err, "Failed to run explain on the query")
+	}
+	defer res.Close()
+
+	explainRes, err := parseExplainResults(res)
+
+	// If our explain parsing fails, we don't analyze the response or fail the query.
+	if err == nil {
```

Comment:
return if err != nil, so we can save an intent?

Reply:
what do you mean "save an intent"?

Reply:
sorry, i mean save a tab... 
Also, we should return err is err != nil

Reply:
Discussed offline, we're going to fail the query if explain parsing fails.